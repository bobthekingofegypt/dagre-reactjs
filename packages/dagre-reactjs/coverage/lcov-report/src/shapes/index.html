
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/shapes</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/shapes</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">66.67% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>28/42</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">69.7% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>23/33</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">83.33% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>5/6</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">61.11% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>22/36</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="Circle.tsx"><a href="Circle.tsx.html">Circle.tsx</a></td>
	<td data-value="87.5" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 87%"></div><div class="cover-empty" style="width: 13%"></div></div>
	</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="77.78" class="pct medium">77.78%</td>
	<td data-value="9" class="abs medium">7/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	</tr>

<tr>
	<td class="file high" data-value="Diamond.tsx"><a href="Diamond.tsx.html">Diamond.tsx</a></td>
	<td data-value="92.86" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 92%"></div><div class="cover-empty" style="width: 8%"></div></div>
	</td>
	<td data-value="92.86" class="pct high">92.86%</td>
	<td data-value="14" class="abs high">13/14</td>
	<td data-value="77.78" class="pct medium">77.78%</td>
	<td data-value="9" class="abs medium">7/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="91.67" class="pct high">91.67%</td>
	<td data-value="12" class="abs high">11/12</td>
	</tr>

<tr>
	<td class="file high" data-value="Rect.tsx"><a href="Rect.tsx.html">Rect.tsx</a></td>
	<td data-value="88.89" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 88%"></div><div class="cover-empty" style="width: 12%"></div></div>
	</td>
	<td data-value="88.89" class="pct high">88.89%</td>
	<td data-value="9" class="abs high">8/9</td>
	<td data-value="69.23" class="pct medium">69.23%</td>
	<td data-value="13" class="abs medium">9/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	</tr>

<tr>
	<td class="file low" data-value="shapes.ts"><a href="shapes.ts.html">shapes.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Wed Mar 24 2021 14:21:11 GMT+0000 (Greenwich Mean Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    