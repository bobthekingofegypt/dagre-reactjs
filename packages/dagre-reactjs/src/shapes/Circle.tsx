import * as React from 'react';

import { ShapeComponentProps } from '../types';

export const Circle: React.FC<ShapeComponentProps> = ({ node, innerSize }) => {
  if (!node || !innerSize || !(innerSize.width && innerSize.height)) {
    return null;
  }
  const r = Math.max(innerSize.width, innerSize.height) / 2;

  return (
    <circle
      style={node.styles.shape.styles || {}}
      x={-innerSize.width / 2}
      y={-innerSize.height / 2}
      r={r}
    ></circle>
  );
};
