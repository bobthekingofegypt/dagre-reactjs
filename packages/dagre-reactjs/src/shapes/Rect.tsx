import * as React from 'react';

import { ShapeComponentProps } from '../types';

export const Rect: React.FC<ShapeComponentProps> = ({ node, innerSize }) => {
  if (!node || !innerSize || !(innerSize.width && innerSize.height)) {
    return null;
  }

  const halfWidth = innerSize.width / 2;
  const halfHeight = innerSize.height / 2;

  return (
    <rect
      data-tip="hello world"
      className={node.styles.shape.className}
      style={node.styles.shape.styles || {}}
      rx={node.styles.shape.cornerRadius || 0}
      ry={node.styles.shape.cornerRadius || 0}
      x={-halfWidth}
      y={-halfHeight}
      width={innerSize.width}
      height={innerSize.height}
    ></rect>
  );
};
