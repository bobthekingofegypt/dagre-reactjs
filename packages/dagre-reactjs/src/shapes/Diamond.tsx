import * as React from 'react';

import { Point, ShapeComponentProps, Size } from '../../src/types';

export const calculateDiamondPoints = (size: Size): Array<Point> => {
  const width = (size.width * Math.SQRT2) / 2;
  const height = (size.height * Math.SQRT2) / 2;

  const points = [
    { x: 0, y: -height },
    { x: -width, y: 0 },
    { x: 0, y: height },
    { x: width, y: 0 },
  ];

  return points;
};

export const Diamond: React.FC<ShapeComponentProps> = ({ node, innerSize }) => {
  if (!node || !innerSize || !(innerSize.width && innerSize.height)) {
    return null;
  }

  const points = calculateDiamondPoints(innerSize);

  return (
    <polygon
      style={node.styles.shape.styles || {}}
      points={points.map((p) => `${p.x}, ${p.y}`).join(' ')}
    />
  );
};
