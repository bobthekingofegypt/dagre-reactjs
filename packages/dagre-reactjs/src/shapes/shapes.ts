import { ShapeDefinition, ShapesDefinition } from '../types';
import { Circle } from './Circle';
import { Diamond } from './Diamond';
import { intersectDiamond } from './intersects/diamond';
import { intersectEllipse } from './intersects/ellipse';
import { intersectRect } from './intersects/rect';
import { Rect } from './Rect';

export const builtInShapes: ShapesDefinition = {
  rect: {
    renderer: Rect,
    intersection: intersectRect,
  },
  circle: {
    renderer: Circle,
    intersection: intersectEllipse,
  },
  diamond: {
    renderer: Diamond,
    intersection: intersectDiamond,
  },
};

export const getShapeDefinition = (
  shape: string,
  shapes: ShapesDefinition
): ShapeDefinition => {
  if (shapes[shape]) {
    return shapes[shape];
  }

  throw Error(`shape ('${shape}') not found in builtins or custom`);
};
