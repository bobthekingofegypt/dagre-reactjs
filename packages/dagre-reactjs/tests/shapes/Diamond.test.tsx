import { shallow } from 'enzyme';
import * as React from 'react';

import { Diamond } from '../../src/shapes/Diamond';
import { getDefaultNodeOptions } from '../stubs';

describe('Diamond', () => {
  it('renders correctly', () => {
    const node = getDefaultNodeOptions('1', 'test label');
    const innerSize = { width: 50, height: 50 };
    const wrapper = shallow(<Diamond node={node} innerSize={innerSize} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('uses the custom stles when passed', () => {
    const node = getDefaultNodeOptions('1', '');
    node.styles.shape.styles = {
      fill: '#000',
    };
    const innerSize = { width: 50, height: 50 };
    const wrapper = shallow(<Diamond node={node} innerSize={innerSize} />);

    const polygon = wrapper.find('polygon');
    expect(polygon.prop('style')).toEqual(node.styles.shape.styles);
  });
});
