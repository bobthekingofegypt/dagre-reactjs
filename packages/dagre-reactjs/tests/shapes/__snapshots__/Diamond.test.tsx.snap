// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diamond renders correctly 1`] = `
<polygon
  points="0, -35.35533905932738 -35.35533905932738, 0 0, 35.35533905932738 35.35533905932738, 0"
  style={
    Object {
      "fillOpacity": 0,
      "stroke": "#000",
    }
  }
/>
`;
